api/src/integrations/nris-emli/datasource.js

```diff
@@ -0,0 +1,466 @@
+          status: 'Completed',
+          message: 'updateRecordType - no records found',
+          itemsProcessed: 0,
+          itemTotal: 0
+        };
+      }
+
+      processingObject.itemTotal = records.length;
+
+      for (let i = 0; i < records.length; i++) {
+        // Make sure these are completed, and >= 7 days before we bring in the record.
+        if (
+          records[i].assessmentStatus === 'Complete' &&
+          moment().diff(moment(records[i].completionDate), 'days') > 7
+        ) {
+          const newRecord = await this.transformRecord(records[i]);
+          const existingRecord = await this.findExistingRecord(newRecord);
+
+          if (existingRecord) {
+            if (newRecord.documents.length === 0) {
+              // create attachment if no existing document was found, if no "Final Report" is attached no document will be created
+              if (this.shouldRecordHaveAttachments(newRecord)) {
+                await this.createRecordAttachments(records[i], newRecord);
+              }
+            }
+
+            await this.updateRecord(newRecord, existingRecord);
+          } else {
+            if (this.shouldRecordHaveAttachments(newRecord)) {
+              await this.createRecordAttachments(records[i], newRecord);
+            }
+            await this.createItem(newRecord);
+          }
+
+          // Assuming we didn't get thrown an error, update the items successfully processed.
+          processingObject.itemsProcessed++;
+        } else {
+          // We skip and don't include non-completed records
+          processingObject.itemTotal--;
+        }
+      }
+      processingObject.status = 'Completed';
+      processingObject.message = 'updateRecordType - all done.';
+    } catch (error) {
+      processingObject.status = 'Failed';
+      processingObject.message = 'updateRecords - unexpected error: ' + error;
+      defaultLog.error(`updateRecords - unexpected error: ${error.message}`);
+    }
+    defaultLog.info('returning', processingObject.status);
+    return processingObject;
+  }
+
+  async transformRecord(record) {
+    const Inspection = mongoose.model(RECORD_TYPE.Inspection._schemaName);
+    let newRecord = new Inspection().toObject();
+    // We don't need this as we insert based on assessmentId
+    delete newRecord._id;
+
+    const legislation = {
+      act: 'Mines Act',
+      section: 15
+    };
+
+    newRecord.legislationDescription = 'Inspection to verify compliance with regulatory requirements.';
+    newRecord.recordType = 'Inspection';
+    newRecord._sourceRefNrisId = record.assessmentId;
+    try {
+      newRecord.dateIssued = record.inspection.inspctReportSentDate
+        ? moment.tz(record.inspection.inspctReportSentDate, 'America/Vancouver').toDate()
+        : null;
+    } catch (error) {
+      defaultLog.debug(error);
+      newRecord.dateIssued = null;
+    }
+    newRecord.issuingAgency = 'Ministry of Energy, Mines and Low Carbon Innovation';
```

Comment:
`issuingAgency` and maybe `author` should be `EMLI` to be consistent with existing DB values.  More details in https://bcmines.atlassian.net/browse/NRPT-361

Reply:
Fixed in 8fa9bd8